libgnucash/engine/mocks/fake-qofquery.h

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file fake-qofquery.h
+ *
+ * @brief Mocking qof queries
+*/
+
+#ifndef FAKE_QOFQUERY_H
+#define FAKE_QOFQUERY_H
+
+#include <gmock/gmock.h>
+
+#include <qofquery.h>
+
+extern "C"
+{
+#include <Query.h>
+}
+
+/** Fake object providing functionality similar to \c QofQuery
+ *
+ * @note \c QofQuery is a \c typedef for \c struct _QofQuery, which is not
+ * public. Therefore class QofFakeQuery is not derived from \c QofQuery.
+ *
+ * To use a QofFakeQuery object simply create it before the GnuCash code
+ * performs a query. Check that the QofFakeQuery object is created with the
+ * correct object type. Also define all expectations and return values on the
+ * created QofFakeQuery object before the GnuCash code performs the query.
+ *
+ * After the query is finished, the QofFakeQuery object can be destroyed.
+ *
+ * @note Internally each created QofFakeQuery object is registered at a
+ * QofFakeQueryPool, which provides it to the GnuCash code on request.
+ * This pool observes the life-cycle of each QofFakeQuery object.
```

Comment:
This would a ggod place to explain that the test will fail with a GTest `ASSERT` in QofFakeQueryPool if `qof_query_create_for()` and `qof_query_destroy()` aren't paired.